pdr_backend/sim/sim_engine.py

```diff
@@ -271,6 +291,51 @@ def save_state(self, i: int, N: int):
         return True, False
 
     @enforce_types
+    def _get_prediction_dataset(
+        self, start_slot: int, end_slot: int
+    ) -> Dict[int, Optional[float]]:
+        contracts = query_feed_contracts(
+            self.ppss.web3_pp.subgraph_url,
+            self.ppss.web3_pp.owner_addrs,
+        )
+
+        sPE = 300 if self.predict_feed.timeframe == "5m" else 3600
+        # Filter contracts with the correct token pair and timeframe
+        contract_to_use = [
+            addr
+            for addr, feed in contracts.items()
+            if feed.symbol
+            == f"{self.predict_feed.pair.base_str}/{self.predict_feed.pair.quote_str}"
+            and feed.seconds_per_epoch == sPE
+        ]
+
+        query = f"""
+            SELECT
+                slot,
+                CASE
+                    WHEN roundSumStakes = 0.0 THEN NULL
+                    WHEN roundSumStakesUp = 0.0 THEN NULL
```

Comment:
for this case when `roundSumStakesUp = 0.0` the value shouldn't be `NULL` because maybe all the predictoors have staked down. It should be `NULL` only if `roundSumStakes = 0.0`